src/codegen/llvm.zig

```diff
@@ -8511,6 +8511,14 @@ pub const FuncGen = struct {
         const dest_ptr = self.sliceOrArrayPtr(dest_slice, ptr_ty);
         const is_volatile = ptr_ty.isVolatilePtr(mod);
 
+        // Any WebAssembly runtime will trap when the destination pointer is out-of-bounds, regardless
+        // of the length. This means we need to emit a check where we skip the memset when the length
+        // is 0 as we allow for undefined pointers in 0-sized slices.
+        const needs_wasm_safety_check = safety and
```

Comment:
Why is `safety` part of the condition here?

Please be careful not to confuse zig safety with avoiding the wasm runtime trapping on memset of length zero. I suggest instead to name this local `intrinsic_len0_traps`.

Reply:
I assumed that when a user used `@setRuntimeSafety(false);` it would include such a check as this. Will change it as per suggestion.

Reply:
The Zig language allows `@memset(undefined, 0);`. There is no illegal behavior here, so there are no safety checks here. Zig safety is an entirely unrelated concept - this is merely a concern with how to lower to target machine code.

Reply:
That makes it all clear to me, thanks!